docs/rules/alphabetical-paths.md

```diff
@@ -0,0 +1,102 @@
+#  Ensure that required paths are in alphabetical order (alphabetical-paths)
+
+## Rule Details
+
+This rule checks whether the required paths in `define` and `require` statements are order alphabetically or not.
+
+### Options
+
+This rule takes two arguments. First argument is optionoal and can be "fullpath", or "basename". If it is "fullpath" then whole path strings are checked for right order. If "basename", then just basename parts of required paths are checked. The default is "fullpath".
+
+The second argument is also optional and will be used only when the first argument is manually set to "fullpath". It can be "always", or "never". If it is "always" then it checks at first just dirnames of paths and then, if they are equal, it checks basenames. If it is "never" then it just checks entire paths. The default is "always".
```

Comment:
> I'm curious why we would need both a "fullpath" and "dirname-then-basename"?

If "dirname-then-basename" (aka "fullpath", "always") will not be used, then this snippet will be invalid with just simple fullpath comparing...

define([
    'aaa/ddd',
    'aaa/bbb/aaa'
], function (ddd, aaa) {
});

Because 'aaa/ddd' < 'aaa/bbb/aaa' = false. But I need it in that order, I need to prefere basedir comparison...and if 2 paths have the same base dir, then compare by their basenames. So I would suggest these options:

```json
{
    "sortBy": "fullpath", // or "basename", or "basedir-basename"
    "ignoreCase": true // or false
}
```

Do you agree?